packages/ui5-config/src/middlewares.ts

```diff
@@ -128,24 +128,53 @@ export function getFioriToolsProxyMiddlewareConfig(
 }
 
 export const getMockServerMiddlewareConfig = (
-    path?: string,
+    appRoot?: string,
+    existingServices: MockserverConfig['services'] = [],
```

Comment:
You offer the optional parameters `existingAnnotations` and `existingServices` however, the only time you call this functions you hardcoded both to `undefined`. So why did you introduce the parameters?
Additionally, you are modifying `existingAnnotations` which could cause side-effects if someone actually calls the function with these parameters.
Lastly, does it make sense to call this function without a `dataSourcesConfig`?

Suggestion: remove the parameters `existingAnnotations` and `existingServices` if you don't have a use case for them.